src/components/Header.astro

```diff
@@ -0,0 +1,114 @@
+---
+import Logo from "./Logo.astro";
+---
+<script>
+let lastScrollTop = 0;
+
+window.addEventListener("scroll", function() {
+    let currentScroll = window.scrollY || document.documentElement.scrollTop;
+    if (currentScroll > lastScrollTop) {
+        // Scroll down
+        document.getElementById("header").classList.add("header-hidden");
+    } else {
+        // Scroll up
+        document.getElementById("header").classList.remove("header-hidden");
+    }
+    lastScrollTop = currentScroll <= 0 ? 0 : currentScroll; // For Mobile or negative scrolling
+}, false);
+
+document.addEventListener("DOMContentLoaded", function() {
+    let currentPage = window.location.pathname; // Get current page filename
```

Comment:
why `let` if you never change this. Please use `npm run lint` to make sure you have no error. I will also add a task to create a pipeline that will automatically do these checks for all your PR's